src/kv/kv_types.h

```diff
@@ -280,6 +280,16 @@ namespace kv
     APPLICATION
   };
 
+  // EntryType must be backwards compatible with the older
+  // bool is_snapshot field
+  static_assert(sizeof(uint8_t) == sizeof(bool));
+
+  enum class EntryType : uint8_t
+  {
+    WriteSet = 0,
+    Snapshot = 1
+  };
```

Comment:
```suggestion
  enum class EntryType : uint8_t
  {
    WriteSet = 0,
    Snapshot = 1
  };
  
  // EntryType must be backwards compatible with the older
  // bool is_snapshot field
  static_assert(sizeof(EntryType) == sizeof(bool));
```

The diff is ugly but moving the assert underneath the enum declaration so we can assert on the size of `EntryType` and not `uint8_t`.

Reply:
That's ok, but we know sizeof(EntryType) == sizeof(uint8_t) because of line 287. What isn't standard and is compiler-dependent is sizeof(bool) however.